### PR TITLE
Removed command line parameters for result filename.

### PR DESCRIPTION
**Pull request by kimjamesuw**
resultFileName is now instantiated inside constructor for XmlRecorder & Hdf5Recorder (will later move to superclass). stateOut_ renamed to resultOut_.

### CODE REVIEW

**Comment by VivekGandhi54** [comment]
The parameter loading still doesn't happen inside `IRecorder::loadParameters()` (No real reason for `IRecorder` to remain an interface class and do this in all derived classes).
It doesn't change much, but might be important/nice for things like issue #216 and #240

**Comment by kimjamesuw** [comment]
> The parameter loading still doesn't happen inside `IRecorder::loadParameters()` (No real reason for `IRecorder` to remain an interface class and do this in all derived classes). It doesn't change much, but might be important/nice for things like issue #216 and #240

I want to make the IRecorder::loadParameters() and some other changes on a different branch, just want to merge these changes for now.

**Comment by stiber** [comment]
> > The parameter loading still doesn't happen inside `IRecorder::loadParameters()` (No real reason for `IRecorder` to remain an interface class and do this in all derived classes). It doesn't change much, but might be important/nice for things like issue #216 and #240
> 
> I want to make the IRecorder::loadParameters() and some other changes on a different branch, just want to merge these changes for now.

This makes sense to me, and of course is a sub-issues of #240. I've created issue #271 for it. Probably makes sense to slot this in once we've designed the new, generic recorder architecture.

File: Simulator/Core/Driver.cpp [hunk 1]
```diff
@@ -196,9 +196,10 @@ bool parseCommandLine(int argc, char *argv[]) {
                     argv[0] + " ");
 
    // Set up the comment line parser.
-   if ((cl.addParam("resultfile", 'o', ParamContainer::filename, "simulation results filepath (deprecated)") !=
-        ParamContainer::errOk)
-       || (cl.addParam("configfile", 'c', ParamContainer::filename,
+   if (
+      //(cl.addParam("resultfile", 'o', ParamContainer::filename, "simulation results filepath (deprecated)") !=
+      //  ParamContainer::errOk) ||
```
Comment (stiber) [inline Simulator/Core/Driver.cpp:201]: Just delete.

**Review by stiber — APPROVED** [review]

**Pull request merged**